### PR TITLE
LPS-133652 Add a case if the record values are of Search Location field

### PR DESCRIPTION
**Pull request by marcelabc**

### CODE REVIEW

**Comment by marcelabc** [comment]
ci:stop

**Comment by marcelabc** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-report-web/src/main/resources/META-INF/resources/js/utils/searchLocation.es.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+			};
+		});
+
+		const newFields = [...fields];
+		const searchLocationFieldIndex = newFields.findIndex(
+			(field) => field.type === 'search_location'
+		);
+		newFields.splice(searchLocationFieldIndex, 1, ...searchLocationFields);
+
+		delete data[searchLocationFieldName];
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-report-web/src/main/resources/META-INF/resources/js/utils/searchLocation.es.js:86]: @marcelabc @EliveltonR18, can you please separate the variable declaration and the following code with a blank line?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/searchLocation/SearchLocationDDMFormFieldValueRenderer.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+		}
+
+		try {
+			JSONObject jsonObject = JSONFactoryUtil.createJSONObject(
+				valueString);
+
+			return jsonObject.getString(fieldName, StringPool.BLANK);
+		}
+		catch (JSONException jsonException) {
+			return StringPool.BLANK;
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/searchLocation/SearchLocationDDMFormFieldValueRenderer.java:69]: @marcelabc, should we return an empty JSON instead? What do you think?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-report-web/src/main/java/com/liferay/dynamic/data/mapping/form/report/web/internal/display/context/DDMFormReportDisplayContext.java [hunk 3]
```diff
@@ -189,6 +193,39 @@ private JSONObject _getDDMFormFieldOptionLabelsJSONObject(
...
 
+	private String _getLabel(DDMFormField ddmFormField) {
+		if (StringUtil.equals(
+				ddmFormField.getType(),
+				DDMFormFieldTypeConstants.SEARCH_LOCATION)) {
+
+			JSONObject jsonObject = JSONFactoryUtil.createJSONObject();
+
+			LocalizedValue localizedValue =
+				(LocalizedValue)ddmFormField.getProperty("visibleFields");
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-report-web/src/main/java/com/liferay/dynamic/data/mapping/form/report/web/internal/display/context/DDMFormReportDisplayContext.java:204]: @marcelabc, shouldn't we rename the variable to `visibleFields` instead?

Actually, this [commit](https://github.com/liferay-data-engine/liferay-portal/pull/567/commits/fea4ad9845d62f7875a1ad26cd8ee5abc23300f7#) of this pull named a similar a variable as `visibleFields`. In light of this, can we rename the 2 variables consistently?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormViewFormInstanceRecordsDisplayContext.java [hunk 4]
```diff
@@ -173,7 +174,8 @@ public String getColumnName(DDMFormField ddmFormField) {
 	}
 
 	public String getColumnValue(
-		DDMFormField ddmFormField, List<DDMFormFieldValue> ddmFormFieldValues) {
+		DDMFormField ddmFormField, List<DDMFormFieldValue> ddmFormFieldValues,
+		String fieldName) {
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormViewFormInstanceRecordsDisplayContext.java:178]: @marcelabc, can we rename `fieldName` to `ddmFormFieldName` to keep consistency with existing back-end code?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/DDMFormFieldValueRenderer.java [hunk 5]
```diff
@@ -25,4 +26,10 @@
 
 	public String render(DDMFormFieldValue ddmFormFieldValue, Locale locale);
 
+	public default String render(
+		DDMFormFieldValue ddmFormFieldValue, String fieldName, Locale locale) {
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/DDMFormFieldValueRenderer.java:30]: @marcelabc, can we rename `fieldName` to `ddmFormFieldName` to keep consistency with existing back-end code?

**Review by natocesarrego — CHANGES_REQUESTED** [review]

**Pull request closed without merging**